benchmarks/NIA/other/sumarraylinear.sl

```diff
@@ -0,0 +1,39 @@
+(set-logic ALIA)
+
+(synth-inv inv_fun ((x (Array Int Int)) (x_length Int) (sum Int) (y Int) (n Int)))
+
+
+(declare-primed-var x (Array Int Int))
+(declare-primed-var x_length Int)
+(declare-primed-var sum Int)
+(declare-primed-var y Int)
+(declare-primed-var n Int)
+
+
+
+(define-fun pre_fun ((x (Array Int Int)) (x_length Int) (sum Int) (y Int) (n Int)) Bool
+  (and (> n 0)
+       (= sum 0)
+       (= y 0)
+       (> x_length 0)
+       (forall ((i Int)) (=> (and (>= i 0) (< i n)) (and (<= 0 i) (< i x_length) (= i (select x i)))))))
+
```

Comment:
Why do we need both `x_length` and `n`? We should just use `n` itself as the length of `x`.

Also, the `x` should be allowed to be empty, i.e. you should have `(>= n 0)` (and change the trans and post accordingly).